subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/ExecuteScheduledTransformationStepBuildOperationType.java

```diff
@@ -27,6 +27,16 @@
 
     public interface Details {
 
+        /**
+         * Returns the unique path of the build this transformation is part of.
+         */
+        String getBuildPath();
+
+        /**
+         * Returns the unique id of the transformation.
+         */
+        long getTransformationId();
+
```

Comment:
Here too should be a single `TransformationIdentifier getIdentifier()` method.

Reply:
IIUC build operations should only use low-level types to avoid serialization breakages. Thus, I think these need to stay two separate properties just like for tasks in `ExecuteTaskBuildOperationType`.

Reply:
How do we do the same thing for tasks?

Reply:
How come we don't use `ExecuteTaskBuildOperationType.Details.getTaskId()` when looking up dependencies of an executed task, but we need to use `ExecuteScheduledTransformationStepBuildOperationType.getTransformationId()` to do the same for transforms?

Reply:
Because we rely on the implementation class of `ExecuteTaskBuildOperationType.Details`, namely `ExecuteTaskBuildOperationDetails`, to get the task instead. I suppose I could do the same thing for transforms.